### ISSUE TITLE
[Bug] Achievement notifications during gameplay although they are disabled in the notification settings

### ISSUE THREAD

**Issue by KingNothing81** [redphx/better-xcloud#383]
### Device

Other

### Operating System

Android

### Browser

Chrome/Edge/Chromium

### Browser Version

124.x

### Better xCloud Version

4.1.0

### Reproduction Steps

Since version 4.1.0 I’m getting achievement pop ups during gameplay again, although I have disabled “Notify me when I unlock an achievement” in the notification settings.

![IMG_0254](https://github.com/redphx/better-xcloud/assets/146887651/740c300c-02d7-4b0d-8486-b6fedcbb96c2)

I think this is related to https://github.com/redphx/better-xcloud/issues/381 and the fix you applied with 4.1.0 to keep the message tray working when “social features” are disabled. The fix works, but it seems that it has created a regression.

I always disliked the achievement pop ups. And was glad that they added a setting to disable them. So if there is a conflict here and it’s not possible to keep both things working, I would really prefer if you revert that change in 4.1.0, so the achievement pop ups are blocked again.

Thanks a lot. And sorry for causing that problem in the first place.

**Comment by redphx** [comment]
I think this is a bug on xCloud's side

**Comment by KingNothing81** [comment]
Hm, I don't think so. As soon as I had discovered that new option in the notification settings, I disabled "Notify me when I unlock an achievement". And never saw an achievment pop up again until I updated to 4.1.0 yesterday. So this would be a very big coincidence.

As I use "disable social features", I think this change was caused by your commit https://github.com/redphx/better-xcloud/commit/ba0ccf5213f7882716e29477889a3678152395a7 . Maybe whatever blocked the notification inbox from working before, was also blocking the achievement pop ups. And then your change in 4.1.0, that made the inbox work again, also unblocked the achievement pop ups again.

If that's really the case, you should make a feature out of this. Because many people also want to get rid of the achievement pop ups. Or maybe you can at least revert the change, As I prefer having a notification inbox that is not working, if this will keep away the achievement pop ups.

**Comment by KingNothing81** [comment]
I had a look at the code of commit https://github.com/redphx/better-xcloud/commit/ba0ccf5213f7882716e29477889a3678152395a7 and saw that with 4.1.0 you stopped blocking "https://notificationinbox.xboxlive.com", but are now blocking "https://xblmessaging.xboxlive.com/network/xbox/users/me/inbox'.

But if blocking "https://notificationinbox.xboxlive.com" really completely disabled the inbox and blocked all notifications (including the pop ups), you should really make an extra option out of this:

A "Do Not Disturb" (DND) mode for xCloud that people can enable. And that can be enabled independently from the "disable social features" option. I think this would be a very useful addition to the script.

**Comment by KingNothing81** [comment]
To further investigate, I turned off “disable social features” to see if it makes a difference. But it doesn’t. I just saw an achievement pop up again.

So you are right. It’s a bug on xCloud‘s side. Showing achievement notifications, although they are disabled in the notification settings.

But I think it’s also true that as long as you blocked “https://notificationinbox.xboxlive.com“, this also blocked the achievement pop ups for me. So I didn’t notice the bug before you unblocked it with 4.1.0. 

This change really feels like a step back for me now. And I hope that you might really consider doing that DND mode. Because I think a feature, that lets users quickly enable/disable getting notifications at all, could be a useful addition for all users.

But as this really turned out to be a xCloud bug, I will close this now.

**Comment by KingNothing81** [comment]
A final update to this:

I used xCloud’s feedback option to report the issue several times now. But so far, it’s still unsolved.

So I created a filter rule in my router to block “https://notificationinbox.xboxlive.com” again. And enabled it for my Shield TV. And now the achievement pop ups are gone again. =)

As I thought, blocking that URL really seems to completely “disable” the inbox by blocking all incoming messages. Including the achievement toast messages.

So although I’ve found a way to solve this problem for me personally, I still think that adding a “DND mode” could also be a useful addition for other users.
